src/Vault.sol

```diff
@@ -910,6 +936,59 @@ contract Vault is ERC4626, ERC20Permit, ILiquidationSource, Ownable {
       _collateralAssets == 0 ? 0 : _shares.mulDiv(_collateralAssets, _depositedAssets, _rounding);
   }
 
+  /**
+   * @notice Convert Vault shares to YieldVault shares.
+   * @param _shares Vault shares to convert
+   * @param _rounding Rounding mode (i.e. down or up)
+   * @return uint256 YieldVault shares
+   */
+  function _convertToYieldVaultShares(
+    uint256 _shares,
+    Math.Rounding _rounding
+  ) internal view returns (uint256) {
+    uint256 _depositedAssets = _totalSupply();
+
+    // When the Vault is collateralized, the yield is excluded from the redeemable amount
+    uint256 _redeemableShares = _isVaultCollateralized()
+      ? _yieldVault.maxRedeem(address(this)) -
+        (_yieldVault.convertToShares(_totalAssets() - _depositedAssets))
+      : _yieldVault.maxRedeem(address(this));
+
+    if (_shares == 0 || _depositedAssets == 0) {
+      return _shares;
+    }
+
+    return
+      _redeemableShares == 0 ? 0 : _shares.mulDiv(_redeemableShares, _depositedAssets, _rounding);
+  }
+
+  /**
+   * @notice Convert Vault assets to YieldVault assets.
+   * @param _assets Vault assets to convert
+   * @param _rounding Rounding mode (i.e. down or up)
+   * @return uint256 YieldVault assets
+   */
+  function _convertToYieldVaultAssets(
+    uint256 _assets,
+    Math.Rounding _rounding
+  ) internal view returns (uint256) {
+    uint256 _depositedAssets = _totalSupply();
+
+    // When the Vault is collateralized, the yield is excluded from the withdrawable amount
+    uint256 _withdrawableAssets = _isVaultCollateralized()
```

Comment:
looks like this is repeating the logic in _collateral()

You could just use that function here instead to calculate the withdrawable collateral